includes/modules/upcoming_products.php

```diff
@@ -22,7 +22,7 @@
 $limit_clause = "  order by " . (EXPECTED_PRODUCTS_FIELD == 'date_expected' ? 'date_expected' : 'products_name') . " " . (EXPECTED_PRODUCTS_SORT == 'asc' ? 'asc' : 'desc') . "
                    limit " . (int)MAX_DISPLAY_UPCOMING_PRODUCTS;
 
-if ( (($manufacturers_id > 0 && $_GET['filter_id'] == 0) || $_GET['music_genre_id'] > 0 || $_GET['record_company_id'] > 0) || (!isset($new_products_category_id) || $new_products_category_id == '0') ) {
+if ( (($manufacturers_id > 0 && $_GET['filter_id'] == 0) || (isset($_GET['music_genre_id']) && $_GET['music_genre_id'] > 0) || (isset($_GET['record_company_id']) && $_GET['record_company_id'] > 0)) || (!isset($new_products_category_id) || $new_products_category_id == '0') ) {
```

Comment:
You can use `! empty($_GET['music_genre_id'])` here instead of the longer `(isset($_GET['music_genre_id']) && $_GET['music_genre_id'] > 0)`

Reply:
I had contemplated that, but wasn't confident that `empty` could correctly be used here.

Based on the one comment, revisited the code to provide an additional commit addressing that one specific comment and other related factors.

This branch currently contains two commits, my original revision to address the issue and then other modifications that have been seen elsewhere (though did not tackle future potential PHP 7.2 related items such as `sizeof` -> `count`).  I have individual modifications on a separate branch though it was chaotic in change this/change that, oh yeah, what about this...

Let me know how if anything should be pulled out/rewritten, etc...

Also, wanted to revise the `while (!$var->EOF)` structure to `foreach ($var as $var_item)` for a few of them, but the `NextRandom()` style method was used and I didn't know how to properly incorporate that other than as provided.

Reply:
Haha! I can appreciate the challenge of wanting to "do everything at once"! 

Since this is against the v155 branch, which ideally is taking only bugfixes, can we reduce this PR down to just what's required to fix the presenting bug, and then do a separate PR to v156 for the broader cleanup?

Reply:
> though did not tackle future potential PHP 7.2 related items such as sizeof -> count.

FWIW the change from sizeof->count isn't a requirement for PHP 7.2. It's just something that shows up in some code review tools where a rule for "avoid alias functions" is enabled.  You'll see that in most cases i reverted back from making that sweeping change.